src/test/txindex_tests.cpp

```diff
@@ -70,6 +71,8 @@ BOOST_FIXTURE_TEST_CASE(txindex_initial_sync, TestChain100Setup)
     }
 
     txindex.Stop(); // Stop thread before calling destructor
+    LOCK(cs_main);
+    txindex_ptr.reset(nullptr);
 }
```

Comment:
Thanks for the txindex test fix. This should be a separate pull request, since the issue is already present (intermittent)

Reply:
Oh, I was wondering how it could be related...